### ISSUE TITLE
UDAs need an experimental flag

### ISSUE THREAD

**Issue by bhalevy** [scylladb/scylla#9664]
Similar to UDFs that are controlled with the `enable_user_defined_functions` config option,
UDAs need a similar option, following the same logic as implemented in `feature_config_from_db_config`
https://github.com/scylladb/scylla/blob/f3d5b2b2b07c036d3074cf1e646da65ae5e84cd1/gms/feature_service.cc#L106-L114

**Comment by bhalevy** [comment]
Cc @tzach

**Comment by psarna** [comment]
I specifically avoided creating yet another flag for UDA, here's the rationale. Creating a user-defined aggregate is absolutely pointless if you haven't created a user-defined function first; I'm not sure if it's even possible to combine current built-in function into a user aggregate, but even if possible, it would not be useful at all - e.g. you could probably create a custom `SUM` or `COUNT` implementation, but why do it if we already have a built-in aggregate for this very purpose. Thus, if somebody wants to use UDA, they need to turn on an UDF experimental flag first anyway.

Also, there's nothing experimental in the implementation of UDA itself - it's an extremely trivial mechanism - it keeps running a function for each row, accumulating the result in a variable, which can then be transformed and returned. I think that yet another additional flag will simply be annoying for users - as in "oh my, I need a rolling restart **again** to use UDA? I just finished doing that for UDF!".

**Comment by nyh** [comment]
I agree with @psarna. Moreover, if we do add a new experimental flag for UDA, let's not make it two flags like we did for UDF (we have both experimental-features=udf and enable_user_defined_functions) - I never understood why we have both.

By the way, if in some point in the future we decide that we want to "graduate" UDF from being experimental but still want UDA to be experimental, we can create the new experimental flag for UDA at that point. We did this in the past with Alternator Streams: At first, CDC was experimental, so Alternator Streams was conditional on the CDC experimental flag. Later, when CDC graduated but Alternator Streams is still considered experimental, Alternator Streams was made conditional on a new ALTERNATOR_STREAMS experimental flag.

**Comment by bhalevy** [comment]
@psarna, @nyh I'm good with using the UDF option to cover UDA as well, and, like Nadav said above, maybe introduce a UDA option in the future when we make UDF not-experimental and if we want to keep UDA as experimental.
